src/base/bittorrent/session.cpp

```diff
@@ -3588,6 +3686,21 @@ const CacheStatus &Session::cacheStatus() const
 void Session::startUpTorrents()
 {
     qDebug("Resuming torrents...");
+    using namespace BitTorrent::DB;
+    QSqlDatabase db = QSqlDatabase::database(MAIN_CONNECTION_NAME);
+    QSqlQuery query(db);
+    query.exec(QLatin1String("PRAGMA user_version"));
+    if (query.next()) {
+        int ver = query.value(0).toInt();
+        if (ver != BitTorrent::DB::VERSION) {
+            LogMsg(tr("Can't load torrents from DB. DB version: %1. Program version: %2").arg(ver).arg(BitTorrent::DB::VERSION),
+                   Log::WARNING);
+            return;
```

Comment:
Backward compatibility must be ensured in some way, otherwise the user will not be able to update the application without the risk of losing their torrents or stopping the application. But this is only a matter of implementation, because we know exactly the structure of the database of previous versions.
However, forward compatibility is generally an unattainable goal, so we must still limit the frequency of database format changes. We need to think about it as carefully as possible.

I had a crazy idea how to provide the possibility of database downgrade that provide some kind of forward compatibility (but "paranoid" will strongly against due to possible security issues). It lies in the fact that we could put SQL queries that perform a database downgrade in some special database table. Then, if a newer database version is found by application, it could retrieve the corresponding SQL queries and execute them.